client/human/HumanClientApp.cpp

```diff
@@ -1054,6 +1058,45 @@ void HumanClientApp::UpdateCombatLogManager() {
 }
 
 namespace {
+    boost::optional<std::string> NewestSinglePlayerSavegame() {
+        using namespace boost::filesystem;
+        try {
+            std::map<std::time_t, path> files_by_write_time;
+
+            auto add_all_savegames_in = [&files_by_write_time](const path& path) {
+                if (!is_directory(path))
+                    return;
+
+                for (directory_iterator dir_it(path);
+                     dir_it != directory_iterator(); ++dir_it)
+                {
+                    const auto& file_path = dir_it->path();
+                    if (!is_regular_file(file_path))
+                        continue;
+                    if (file_path.extension() != SP_SAVE_FILE_EXTENSION)
+                        continue;
+
+                    std::time_t t = last_write_time(file_path);
+                    files_by_write_time.insert({t, file_path});
+                }
+            };
+
+            // Find all save games in either player or autosaves
+            add_all_savegames_in(GetSaveDir());
+            add_all_savegames_in(GetSaveDir() / "auto");
```

Comment:
An issue with this bit is that it only looks in the root save dir and root/auto, not any other save directory subdirectories. This means that if a player wants to use directories to organize their saves, they can't. Previously one could open the load dialog and look into the subdirectories (or random other places, but that's less important).